src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java

```diff
@@ -152,55 +174,94 @@ private TaskHandlerImpl(ActivityTaskHandler handler) {
 
     @Override
     public void handle(
-        IWorkflowService service, String domain, String taskList, PollForActivityTaskResponse task)
+        IWorkflowService service, String domain, String taskList, MeasurableActivityTask task)
         throws Exception {
+      options
+          .getMetricsScope()
+          .timer(MetricsType.TASK_LIST_QUEUE_LATENCY)
+          .record(
+              Duration.ofNanos(
+                  task.task.getStartedTimestamp() - task.task.getScheduledTimestamp()));
+
       try {
-        ActivityTaskHandler.Result response = handler.handle(service, domain, task);
-        sendReply(task, response);
+        Stopwatch sw = options.getMetricsScope().timer(MetricsType.ACTIVITY_EXEC_LATENCY).start();
+        ActivityTaskHandler.Result response =
+            handler.handle(service, domain, task.task, options.getMetricsScope());
+        sw.stop();
+
+        sw = options.getMetricsScope().timer(MetricsType.ACTIVITY_RESP_LATENCY).start();
+        sendReply(task.task, response);
+        sw.stop();
```

Comment:
Shouldn't we have a separate counter for the sending reply failures?

Reply:
client has it per each rpc call. If we think we need it here we can add another one.